intel_extension_for_transformers/llm/runtime/graph/models/gptneox/gptneox_utils.cpp

```diff
@@ -193,3 +181,113 @@ class gptneox_quant_layer : public quant_layer_base {
   }
 };
 REGISTER_QUANT_LAYER_CLASS(gptneox);
+
+class gptneox_beam_search_kv_cache_reorder : public beam_search_kv_cache_reorder {
+ public:
+  explicit gptneox_beam_search_kv_cache_reorder(model_context* lctx) : beam_search_kv_cache_reorder(lctx) {}
+  ~gptneox_beam_search_kv_cache_reorder() {}
+
+  virtual void update(const uint32_t& n_past, const uint32_t& n_prompt_tokens,
+                      const std::vector<std::tuple<int, int>>& kv_reorder_indices = {},
+                      const std::vector<beam>& next_beams = {}) override {
+    // first step
+    if (n_past == n_prompt_tokens) {
+      // cpy batch 1 to all batches
+#pragma omp parallel for
+      for (int i = 0; i < ctx->model.layers.size(); ++i) {
```

Comment:
usually layers size is much less than core numbers, so can we do more parallel？

@DDEle cc our expert for any inputs

Reply:
can we try what we do in previous concat optimization?

https://github.com/oneapi-src/oneDNN/blob/34ad8b755b1af7251177d362d1345c732cf4fabc/src/cpu/simple_concat.cpp#L125

or

https://github.com/intel/intel-extension-for-transformers/blob/4f5de037f524de40839ccc524664f98f2937d711/intel_extension_for_transformers/llm/runtime/deprecated/executor/src/operators/concat.cpp#L267

Reply:
https://github.com/intel/intel-extension-for-transformers/pull/449/commits/c0a27307c95eeb75dfbcfe8ad71c14d505f1a44a

Reply:
though not quite ideal but still a good start, you can keep optimizing if you want.